typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -0,0 +1,169 @@
+import debug from 'debug';
+
+import {
+  OptimismISM,
+  OptimismISM__factory,
+  OptimismMessageHook,
+  OptimismMessageHook__factory,
+  TestRecipient,
+  TestRecipient__factory,
+} from '@hyperlane-xyz/core';
+import { types } from '@hyperlane-xyz/utils';
+import { addressToBytes32 } from '@hyperlane-xyz/utils/dist/src/utils';
+
+import { HyperlaneContracts, HyperlaneContractsMap } from '../contracts';
+import { HyperlaneDeployer } from '../deploy/HyperlaneDeployer';
+import { MultiProvider } from '../providers/MultiProvider';
+import { ChainMap, ChainName } from '../types';
+
+import { isHookConfig, isISMConfig } from './config';
+import { HookFactories, hookFactories } from './contracts';
+import { HookConfig } from './types';
+
+export class HyperlaneHookDeployer extends HyperlaneDeployer<
+  HookConfig,
+  HookFactories
+> {
+  constructor(multiProvider: MultiProvider) {
+    super(multiProvider, hookFactories, {
+      logger: debug('hyperlane:HookDeployer'),
+    });
+  }
+
+  async deploy(
+    configMap: ChainMap<HookConfig>,
+  ): Promise<HyperlaneContractsMap<HookFactories>> {
+    let ismContracts: HyperlaneContracts<HookFactories> | undefined;
+    let hookContracts: HyperlaneContracts<HookFactories> | undefined;
+
+    // Process ISM configs first
+    for (const [chain, config] of Object.entries(configMap)) {
+      if (isISMConfig(config)) {
+        ismContracts = await this.deployContracts(chain, config);
+      }
+    }
+
+    // Ensure ISM contracts have been deployed
+    if (!ismContracts || !ismContracts?.optimismISM) {
+      throw new Error('ISM contracts not deployed');
+    }
+
+    // Then process hook configs
+    for (const [chain, config] of Object.entries(configMap)) {
+      if (isHookConfig(config)) {
+        config.remoteIsm = ismContracts.optimismISM.address;
+        this.logger(`Remote ISM address set as ${config.remoteIsm}`);
+        hookContracts = await this.deployContracts(chain, config);
+      }
+    }
+
+    // Ensure hook contracts have been deployed
+    if (!hookContracts || !hookContracts?.optimismMessageHook) {
+      throw new Error('Hook contracts not deployed');
+    }
```

Comment:
you handle multiple networks of config but only handle a single `hookContracts` and `ismContracts`